lib/TcpConnection.js

```diff
@@ -218,7 +218,10 @@ module.exports = function factory (userOptions) {
         }
       })
     })
-    retry.on('fail', (err) => process.stderr.write(`could not reconnect: ${err.message}`))
+    retry.on('fail', (err) => {
+      process.stderr.write(`could not reconnect${err && `: ${err.message}`}`)
```

Comment:
I would like to remove this log in favor of an event `reconnectFailure` (see below)

Reply:
Yes, please remove.